packages/components/src/Modal/Modal.rebuilt.tsx

```diff
@@ -0,0 +1,155 @@
+import { Button } from "../Button";
+import { AtlantisPortalContent } from "../AtlantisPortalContent";
+
+export function ModalHeader({ title, children }: HeaderProps) {
+  const { header, dismissButton } = useModalStyles();
+  const { dismissible, onRequestClose, modalLabelledBy } = useModalContext();
+
+  if (children) {
+    return <>{children}</>;
+  }
+
+  return (
+    <div className={header} data-testid="ATL-Modal-Header" id={modalLabelledBy}>
+      <Heading level={2}>{title}</Heading>
+
+      {dismissible && (
+        <div className={dismissButton}>
+          <ButtonDismiss onClick={onRequestClose} ariaLabel="Close modal" />
+        </div>
+      )}
+    </div>
+  );
+}
+
+export function ModalActions({
+  primary,
+  secondary,
+  tertiary,
+}: ModalActionsProps) {
+  const { actionBar, rightAction, leftAction } = useModalStyles();
+  const shouldShow =
+    primary != undefined || secondary != undefined || tertiary != undefined;
+
+  return (
+    <>
+      {shouldShow && (
+        <div className={actionBar} data-testid="ATL-Modal-Actions">
+          <div className={rightAction}>
+            {primary && <Button {...primary} />}
+            {secondary && (
+              <Button {...secondary} type="primary" variation="subtle" />
+            )}
+          </div>
+          {tertiary && (
+            <div className={leftAction}>
+              <Button {...tertiary} type="secondary" variation="destructive" />
+            </div>
+          )}
+        </div>
+      )}
+    </>
+  );
+}
+
+export function ModalActivator({ children }: PropsWithChildren) {
+  const { activatorRef } = useModalContext();
+
+  return (
+    <span ref={activatorRef} style={{ display: "contents" }}>
+      {children}
+    </span>
+  );
+}
+
+/**
+ * Background overlay for the modal. Used in the ModalContent.
+ */
+
+export function ModalOverlay() {
+  const { onRequestClose } = useModalContext();
+  const { overlayBackground } = useModalStyles();
+
+  return (
+    <motion.div
+      onClick={onRequestClose}
+      className={overlayBackground}
+      initial={{ opacity: 0 }}
+      animate={{ opacity: 0.8 }}
+      exit={{ opacity: 0 }}
+      transition={{ duration: 0.2 }}
+    />
+  );
+}
+
+export function ModalContent({ children }: ModalContainerProps) {
+  const {
+    open,
+    floatingContext,
+    activatorRef,
+    floatingRefs,
+    size,
+    floatingNodeId,
+    modalLabelledBy,
+  } = useModalContext();
+  const { modal, overlay } = useModalStyles(size);
+
+  return (
+    <AnimatePresence>
+      {open && (
+        <FloatingNode id={floatingNodeId}>
+          <FloatingPortal>
+            <AtlantisPortalContent>
+              <FloatingOverlay className={overlay}>
+                <FloatingFocusManager
+                  context={floatingContext}
+                  returnFocus={activatorRef?.current ? activatorRef : true}
+                  initialFocus={floatingRefs?.floating}
+                >
+                  <div
```

Comment:
Moved this because I found that the overlay was slightly changed unintentionally, this better matches the dom structure of the original modal